qiita_db/handlers/analysis.py

```diff
@@ -56,15 +59,34 @@ def get(self, analysis_id):
         -------
         dict
             The contents of the analysis keyed by sample id
+
+        Notes
+        -----
+        This response needed to be broken in chunks because we were hitting
+        the max size of a respose: 2G; based on: https://bit.ly/3CPvyjd
         """
+        chunk_len = 1024 * 1024 * 1  # 1 MiB
+
         with qdb.sql_connection.TRN:
             a = _get_analysis(analysis_id)
             mf_fp = qdb.util.get_filepath_information(
                 a.mapping_file)['fullpath']
-            response = None
             if mf_fp is not None:
                 df = qdb.metadata_template.util.load_template_to_dataframe(
                     mf_fp, index='#SampleID')
-                response = df.to_dict(orient='index')
+                response = dumps(df.to_dict(orient='index'))
```

Comment:
i think all of this makes sense, but I think it should be possible to do this outside of the TRN context, and close the transaction?